services/QuillLMS/spec/support/shared_contexts/snapshots/snapshots_cte.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+RSpec.shared_context 'Snapshots CTE' do
+  # Create before running the query, otherwise, some parameters (e.g. school_ids) will be nil
+  before { cte_records }
+
+  let(:results) { QuillBigQuery::TestRunner.run(described_class.new(*params).final_query, cte_records) }
```

Comment:
I wonder if it would be better to restructure this a bit to still call `run`, but maybe making the runner a variable that we can stub in testing. This might make it easier to stub this out in other tests, and it adds a clean seam for testing. e.g.
```ruby
let(:runner) { QuillBigQuery::TestRunner.new(cte_records) }
let(:results) {described_class.run(*params, runner: runner) }

# Query class
def new(params, runner: QuillBigQuery::Runner)
  # initialization
end

def run
  runner.execute(query)
end

Reply:
My concern with the current setup is we are stubbing everything around a specific sub-method, and not using the path of the code used by the application.